dcp-0008/dcp-0008.mediawiki

```diff
@@ -0,0 +1,341 @@
+with a given version doesn't guarantee it actually adheres to rules that apply
+to that version since it might have been created prior to the definition of
+those rules.
+
+The changes specified by this proposal permit ecosystem components to safely and
+reliably associate a given version with a specific set of well-defined consensus
+rules.
+
+==Specification==
+
+All transactions and scripts with a version that is greater than their currently
+supported maximum values MUST be rejected until a future consensus rule change
+introduces a new version along with fully defining its semantics.
+
+As of the implementation of this specification, this means imposing the
+following two new rules:
+
+* All transactions with versions greater than 3 MUST be rejected
+* All scripts with versions greater than 0 MUST be rejected
+
+<img src="explicit_version_upgrades.svg" width="100%" height="492" />
+
+==Rationale==
+
+The approach taken for these changes was chosen for the following primary reasons:
+
+* Nearly every ecosystem component must work with transactions and/or scripts in one way or another implying they necessarily have access to the transaction version as well as the script version for each of its outputs
+* The transaction and script version fields are self-contained values that do not rely on global state
+* Rejecting newer versions ensures ecosystem components encountering a specific version can safely and reliably associate specific rules with it
+* It makes the upgrade process for new transaction and script versions require hard forks
+
+More generally, there is a strong preference for changes which do not require
+permanently carrying around legacy baggage that ultimately can't be fixed.  The
+proposed implementation satisfies that goal.
+
+===Hard Forks vs Soft Forks and Backwards Compatibility===
+
+An extremely widespread, and astonishingly inaccurate, claim made in regards to
+soft forks vs hard forks is the claim that soft forks are backwards compatible
+while hard forks are not.  It is important to note that this claim is entirely
+incorrect.  In fact, '''backwards compatibility is an absolute requirement '''
+'''for all consensus upgrades regardless of which method is used.'''
+
+Backwards compatibility is the ability for newer systems to interoperate with
+older systems.  When it comes to cryptocurrencies and full nodes, regardless of
+whether a hard fork or soft fork is used for consensus upgrades, all newer
+versions of the software must necessarily be backwards compatible or they
+would not be able to fully validate historical blocks nor would it be possible
+to spend coins originally created in older versions.
+
+What soft forks actually provide that hard forks do not is
+'''forwards compatibility'''.  That is to say the ability of older systems to
+accept input created by newer versions of the system.  In the context of
+cryptocurrencies, and Decred in particular, this would mean older versions of
```

Comment:
```suggestion
cryptocurrencies and Decred in particular, this would mean older versions of
```
no need for comma

Reply:
I believe the comma is required because it's a [non-restrictive appositive](https://en.wikipedia.org/wiki/Apposition).

Reply:
ok